### ISSUE TITLE
Make Julia dependency optional with a pure Python simulator? 

### ISSUE THREAD

**Issue by pierre-haessig** [BondGraphTools/BondGraphTools#131]
Hello,

I just came across your package while searching for Python tools for bond-graphs. I'm starting a new course with a bit of bond graphs in the syllabus.

Besides Python, I'm also using Julia, so I'm not that bothered by the Julia dependency, except that it's the legacy 0.6 version. My main concern would be if I want to use it with a group of undergraduate students working on their own computer.

So my question is: would it be possible to implement a pure Python simulator (based on [scipy.integrate](https://docs.scipy.org/doc/scipy/reference/integrate.html)) so that the dependency on Julia would become optional? Of course simulation performance would be lower, but for small toy models, this shouldn't be a problem.

What do you think?

For the reference, this question may have some links with the open issue of diffeqpy migration #124

**Comment by peter-cudmore** [comment]
Hi Pierre.
A pure python simulator would definitely be ideal.

**Comment by peter-cudmore** [comment]
scikit-odes has sundials bindings..

https://github.com/bmcage/odes

We should port the simulation bindings to use this instead of Julia.

**Comment by pierre-haessig** [comment]
OK, I had not in mind the ODE vs DAE issue. I've never used `scikit.odes`, but it sounds like a reasonable dependency. Their README claims a Python 3.3-3.5 compatibility which looks a bit outdated, but in [common.py](https://github.com/bmcage/odes/blob/master/common.py) which feeds the `setup.py`, it's 3.5 - 3.7, so it's pretty up to date, with several fresh releases.

On my side, I've finished the Bond Graph course, and I ended up not doing BG simulations. I'll revisit this for the next iteration in November.
